homeworks/hw12/src/bank.rs

```diff
@@ -0,0 +1,869 @@
+                    let operation = Operation {
+                        id: transaction_id.to_owned(),
+                        source_account: sender_account.to_owned(),
+                        amount,
+                        target_account: Some(receiver_account.to_owned()),
+                        operation_type: OperationType::Transfer,
+                    };
+                    self.push_transaction(operation);
+                    info!(
+                        "Transaction id: {} Transferred {} from {} to {}",
+                        transaction_id, amount, sender_account, receiver_account
+                    );
+                    Ok(transaction_id.to_owned())
+                }
+            } else {
+                Err(AccountNotFoundError {
+                    account: receiver_account.to_owned(),
+                }
+                .into())
+            }
+        } else {
+            Err(AccountNotFoundError {
+                account: sender_account.to_owned(),
+            }
+            .into())
+        }
+    }
+    /// Returns the current balance of the account.
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    /// The current balance of the account.
+    /// # Errors
+    /// AccountNotFoundError
+    /// ```
+    fn get_balance(&self, account: &str) -> Result<Money, BankError> {
+        debug!("get_balance  for account {}", account);
+        if self.accounts.contains_key(account) {
+            Ok(self
+                .accounts
+                .get(account)
+                .map(|balance| *balance.borrow())
+                .unwrap())
+        } else {
+            Err(AccountNotFoundError {
+                account: account.to_owned(),
+            }
+            .into())
+        }
+    }
+    /// Returns the transaction history of the specified account.
+    ///
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    ///
+    /// A vector of strings representing the transaction history of the account.
+    ///
+    /// # Errors
+    /// BankError
+    /// Returns an error if the specified account does not exist.
+    /// ```
+    fn get_history(&self) -> Result<Vec<Operation>, BankError> {
+        // Ok(self.history.iter().cloned().collect())
+        Ok(self.history.values().cloned().collect())
+    }
+    /// Returns the transaction history of the specified account.
+    ///
+    /// # Arguments
+    ///
+    /// * `account` - The name of the account for which to retrieve the transaction history.
+    ///
+    /// # Returns
+    ///
+    /// A vector of strings representing the transaction history of the account.
+    ///
+    /// # Errors
+    /// BankError
+    /// ```
+
+    fn get_account_history(&self, account: &str) -> Result<Vec<Operation>, BankError> {
```

Comment:
```suggestion
    /// ```
    fn get_account_history(&self, account: &str) -> Result<Vec<Operation>, BankError> {
```